diff --git a/lib/dvb/demux.cpp b/lib/dvb/demux.cpp
index 0631c5f..1cb6e07 100644
--- a/lib/dvb/demux.cpp
+++ b/lib/dvb/demux.cpp
@@ -112,7 +112,16 @@ RESULT eDVBDemux::setSourceFrontend(int fenum)
 	int n = DMX_SOURCE_FRONT0 + fenum;
 	int res = ::ioctl(fd, DMX_SET_SOURCE, &n);
 	if (res)
+	{
 		eDebug("[eDVBDemux] DMX_SET_SOURCE Frontend%d failed: %m", fenum);
+#if HAVE_AMLOGIC
+		/** FIXME: gg begin dirty hack  */
+		eDebug("[eDVBDemux] Ignoring due to limitation to one frontend for each adapter and missing ioctl....");
+		source = fenum;
+		res = 0;
+		/** FIXME: gg end dirty hack  */
+#endif
+	}
 	else
 		source = fenum;
 	::close(fd);
