diff -Naur servicemp3-PLi/servicemp3/servicemp3.cpp servicemp3-PersianPrince/servicemp3/servicemp3.cpp
--- servicemp3-PLi/servicemp3/servicemp3.cpp	2019-01-20 17:06:12.000000000 +0330
+++ servicemp3-PersianPrince/servicemp3/servicemp3.cpp	2019-02-11 12:25:27.528089601 +0330
@@ -21,6 +21,10 @@
 #include <gst/pbutils/missing-plugins.h>
 #include <sys/stat.h>
 
+extern "C" {
+#include <codec.h>
+}
+
 #define HTTP_TIMEOUT 30
 
 /*
@@ -624,6 +628,7 @@
 #endif
 	if ( m_gst_playbin )
 	{
+		aml_set_mediaplay_source((void *)m_gst_playbin,(int)m_sourceinfo.is_audio);
 		/*
 		 * avoid video conversion, let the dvbmediasink handle that using native video flag
 		 * volume control is done by hardware, do not use soft volume flag
